tools/vscode/ddlog-language/src/syntax/entity.json5

```diff
@@ -0,0 +1,84 @@
+{
+  "repository": {
+        "entity": {
+            "patterns": [
+                {
+                    "match": "(\\b)(?<=relation)(\\s)([A-Z0-9_][a-zA-Z0-9_]+)",
+                    "name": "entity.name.relation.ddlog"
+                },
+                {
+                    "match": "(\\b)(?<=relation)(\\s)([A-Z0-9_]*)",
+                    "name": "entity.name.relation.ddlog"
+                },                             
+                {
+                    "match": "\\b(?<=function)(\\b)([a-z_][a-zA-Z0-9_]*)",
+                    "name": "entity.name.function.ddlog",
+                    "captures": {
+                        "1": {
+                            "name": "keyword.operator.group.parameters.ddlog"
+                        }
+                    }
+                },
+                {
+                    "match": "(\\b)([a-z_][a-zA-Z0-9_]+)(?=[(])",
+                    "name": "entity.name.function.ddlog",
+                    "captures": {
+                        "1": {
+                            "name": "keyword.operator.group.parameters.ddlog"
+                        }
+                    }
+                },
+                {
+                    "match": "(\\b)([a-z_]+)(?=[(])",
+                    "name": "entity.name.function.ddlog",
+                    "captures": {
+                        "1": {
+                            "name": "keyword.operator.group.parameters.ddlog"
+                        }
+                    }
+                },
+                {
+                    "match": "(?<=')([A-Z][a-zA-Z0-9_]*)",
+                    "name": "entity.name.type-variable.ddlog"
+                },
+                {
+                    "match": "(?<=(\\=))(\\s*)([A-Z][a-zA-Z0-9_]*)",
+                    "name": "entity.name.constructor.ddlog"
+                },
+                {
+                    "match": "(?<=(\\|))(\\s*)([A-Z][a-zA-Z0-9_]*)",
```

Comment:
| exists both as a type constructor and as a binary expression. I hope this is used within the right context, but I had the impression that the matching of | as a constructor alternative does not work very well.